modules/arrow-flight-rpc/build.gradle

```diff
@@ -0,0 +1,259 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+apply plugin: 'opensearch.publish'
+apply plugin: 'opensearch.internal-cluster-test'
+
+opensearchplugin {
+  description 'Arrow flight based Stream implementation'
+  classname 'org.opensearch.arrow.flight.FlightStreamPlugin'
+}
+
+dependencies {
+  implementation project(':libs:opensearch-arrow-spi')
+
+  implementation "io.netty:netty-buffer:${versions.netty}"
```

Comment:
Did you consider using the [netty shaded gRPC dependency](https://github.com/opensearch-project/OpenSearch/blob/main/plugins/transport-grpc/build.gradle#L23)? The gRPC maintainers seem to recommend using that as they sometimes depend on unstable Netty APIs and the shaded version avoids problems on upgrades.

Reply:
good point, let me try

Reply:
i have shaded them under opensearch namespace as arrow-memory just used netty-buffer and netty-common. Check `libs/arrow-memory-shaded`.  I have shaded other commons dependencies as well like `slf4j`, `flatbuffers`, `jackson` and `commons.codec` to prevent pollution of runtime classpath of server.

Reply:
We got rid of all shaded dependencies in the latest revision. We will bring them back when arrow will be used in server, right now its contained to plugin `arrow-flight-rpc`.